frontend/src/app/core/services/table-export.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import * as FileSaver from 'file-saver';
+import * as XLSXManager from 'xlsx';
+import jsPDF from 'jspdf';
+import autoTable, { Styles } from 'jspdf-autotable';
+
+const JsPDF = jsPDF;
+
+@Injectable({
+    providedIn: 'root'
+})
+export class TableExportService {
+    private static saveAsExcelFile(buffer: BlobPart[], fileName: string, type: string, extension: string): void {
+        const data: Blob = new Blob(buffer, { type });
+        FileSaver.saveAs(data, `${fileName}.${extension}`);
+    }
+
+    exportPdf(arrayOfItemsToWrite: any[], fileName: string, orientation: 'landscape' | 'portrait' = 'landscape'): void {
+        const doc = new JsPDF({ orientation, unit: 'px', compress: true });
+        if (arrayOfItemsToWrite.length) {
+            const cols: string[][] = [Object.keys(arrayOfItemsToWrite[0])];
+            const rows: string[][] = arrayOfItemsToWrite.map(item => Object.values(item));
+
+            const stylesPerProperty: { [p: string]: Partial<Styles> } = {};
+            cols[0].forEach((propName, index) => {
+                stylesPerProperty[index] = { minCellWidth: propName.length * 8 };
+            });
```

Comment:
Why you cannot use a map here?
And please remove "magic number" - create a const for it

Reply:
I need `index` here. So i will read a little bit of doc about map, maybe i can use here a `map` with index

Reply:
This binds index (of property) to his styles. I don't think that i need here a map

Reply:
Autotable wants data like `arr[5] = { style: 123 }`

Reply:
I don't need array, i need a key-value where key is `index` and value is `Partial<Styles>`